src/common/mesh_utils.cpp

```diff
@@ -50,4 +50,108 @@ std::shared_ptr<mfem::ParMesh> buildParallelMesh(const std::string& mesh_file, c
+  mfem::Vector vertex(dim);
+  for (int i = 0; i < num_vertices; i++) {
+    for (int d = 0; d < dim; d++) {
+      vertex(d) = vertices[d * num_vertices + i];
+    }
+
+    double L1_norm = vertex.Norml1();
+    double L2_norm = vertex.Norml2();
+    vertex *= (L2_norm < 1.0e-6) ? 0.0 : (L1_norm / L2_norm);
+
+    for (int d = 0; d < dim; d++) {
+      vertices[d * num_vertices + i] = vertex(d);
+    }
+  }
+  mesh.SetVertices(vertices);
+}
+
+std::unique_ptr<mfem::Mesh> DiskMesh(int approx_number_of_elements)
+{
+  static constexpr int dim                   = 2;
+  static constexpr int num_elems             = 4;
+  static constexpr int num_vertices          = 5;
+  static constexpr int num_boundary_elements = 4;
+
+  static constexpr double vertices[num_vertices][dim] = {{0, 0}, {1, 0}, {0, 1}, {-1, 0}, {0, -1}};
+  static constexpr int    triangles[num_elems][3]     = {{1, 2, 0}, {2, 3, 0}, {3, 4, 0}, {4, 1, 0}};
+  static constexpr int    segments[num_elems][2]      = {{1, 2}, {2, 3}, {3, 4}, {4, 1}};
+
+  auto mesh = std::make_unique<mfem::Mesh>(dim, num_vertices, num_elems, num_boundary_elements);
+
+  for (auto vertex : vertices) {
+    mesh->AddVertex(vertex);
+  }
+  for (auto triangle : triangles) {
+    mesh->AddTriangle(triangle);
+  }
+  for (auto segment : segments) {
+    mesh->AddBdrSegment(segment);
+  }
+  mesh->FinalizeTriMesh();
+
+  while (mesh->GetNE() < (0.5 * approx_number_of_elements)) {
+    mesh->UniformRefinement();
+  }
+
+  squish(*mesh);
+
+  return mesh;
+}
+
+std::unique_ptr<mfem::Mesh> BallMesh(int approx_number_of_elements)
+{
+  static constexpr int dim                   = 3;
+  static constexpr int num_elems             = 8;
+  static constexpr int num_vertices          = 7;
+  static constexpr int num_boundary_elements = 8;
+
+  static constexpr double vertices[num_vertices][dim] = {{0, 0, 0}, {-1, 0, 0}, {0, 1, 0}, {0, 0, -1},
+                                                         {0, 0, 1}, {0, -1, 0}, {1, 0, 0}};
+  static constexpr int    triangles[num_elems][3]     = {{4, 5, 6}, {4, 6, 2}, {4, 2, 1}, {4, 1, 5},
+                                                  {5, 1, 3}, {5, 3, 6}, {3, 1, 2}, {6, 3, 2}};
+  static constexpr int    tetrahedra[num_elems][4]    = {{0, 4, 5, 6}, {0, 4, 6, 2}, {0, 4, 2, 1}, {0, 4, 1, 5},
+                                                   {0, 5, 1, 3}, {0, 5, 3, 6}, {0, 3, 1, 2}, {0, 6, 3, 2}};
+
+  auto mesh = std::make_unique<mfem::Mesh>(dim, num_vertices, num_elems, num_boundary_elements);
+
+  for (auto vertex : vertices) {
+    mesh->AddVertex(vertex);
+  }
+  for (auto tetrahedron : tetrahedra) {
+    mesh->AddTet(tetrahedron);
+  }
+  for (auto triangle : triangles) {
+    mesh->AddBdrTriangle(triangle);
+  }
+  mesh->FinalizeTetMesh();
+
+  while (mesh->GetNE() < (0.25 * approx_number_of_elements)) {
+    mesh->UniformRefinement();
+  }
+
+  squish(*mesh);
+
+  return mesh;
+}
+
+std::unique_ptr<mfem::Mesh> RectangleMesh(int elements_in_x, int elements_in_y)
```

Comment:
Another thought - it would make the user code cleaner if these calls took an MPI communicator and returned a parallel mesh (similar to [`buildParallelMesh`](https://github.com/LLNL/serac/blob/8b9a6c9698404744bb459845dc8d5d2e0b26e535/src/common/mesh_utils.hpp#L35)).

Reply:
Maybe call that function `buildMeshFromFile` or something and these functions `buildRectangleMesh`.

Reply:
Should we have parallel and non-parallel mesh versions?

Reply:
No. We require the MPI-ified mfem and therefore we need full mfem parallel meshes for all applications.